src/transformers/models/canine/modeling_canine.py

```diff
@@ -0,0 +1,1559 @@
+            if attention_mask.ndim == 3:
+                # if attention_mask is 3D, do the following:
+                attention_mask = torch.unsqueeze(attention_mask, dim=1)
+                # Since attention_mask is 1.0 for positions we want to attend and 0.0 for
+                # masked positions, this operation will create a tensor which is 0.0 for
+                # positions we want to attend and -10000.0 for masked positions.
+                attention_mask = (1.0 - attention_mask.float()) * -10000.0
+            # Apply the attention mask (precomputed for all layers in CanineModel forward() function)
+            attention_scores = attention_scores + attention_mask
+
+        # Normalize the attention scores to probabilities.
+        attention_probs = nn.Softmax(dim=-1)(attention_scores)
+
+        # This is actually dropping out entire tokens to attend to, which might
+        # seem a bit unusual, but is taken from the original Transformer paper.
+        attention_probs = self.dropout(attention_probs)
+
+        # Mask heads if we want to
+        if head_mask is not None:
+            attention_probs = attention_probs * head_mask
+
+        context_layer = torch.matmul(attention_probs, value_layer)
+
+        context_layer = context_layer.permute(0, 2, 1, 3).contiguous()
+        new_context_layer_shape = context_layer.size()[:-2] + (self.all_head_size,)
+        context_layer = context_layer.view(*new_context_layer_shape)
+
+        outputs = (context_layer, attention_probs) if output_attentions else (context_layer,)
+
+        return outputs
+
+
+class CanineSelfOutput(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.dense = nn.Linear(config.hidden_size, config.hidden_size)
+        self.LayerNorm = nn.LayerNorm(config.hidden_size, eps=config.layer_norm_eps)
+        self.dropout = nn.Dropout(config.hidden_dropout_prob)
+
+    def forward(self, hidden_states, input_tensor):
+        hidden_states = self.dense(hidden_states)
+        hidden_states = self.dropout(hidden_states)
+        hidden_states = self.LayerNorm(hidden_states + input_tensor)
+        return hidden_states
+
+
+class CanineAttention(nn.Module):
+    """
+    Additional arguments related to local attention:
+
+        local: Whether to apply local attention. always_attend_to_first_position: Should all blocks be able to attend
+        to the `to_tensor`'s first position (e.g. a [CLS] position)? first_position_attends_to_all: Should the
+        `from_tensor`'s first position be able to attend to all positions within the `from_tensor`?
+        attend_from_chunk_width: The width of each block-wise chunk in `from_tensor`. attend_from_chunk_stride: The
+        number of elements to skip when moving to the next block in `from_tensor`. attend_to_chunk_width: The width of
+        each block-wise chunk in `to_tensor`. attend_to_chunk_stride: The number of elements to skip when moving to the
+        next block in `to_tensor`.
```

Comment:
This won't work for sphinx. The syntax used in [configuration_utils](https://github.com/huggingface/transformers/blob/a79585bbf97c0eb23265aaa1bb8043d387f48992/src/transformers/configuration_utils.py#L97) should be followed for new arguments like this otherwise it won't render properly in the doc and the doc_styling script is going to do weird things like here.

Reply:
Ok, but the docs of `CanineAttention` won't show up anywhere in HTML pages, it's just internal documentation. Not sure whether I should make the parameter names bold, like in `configuration_utils.py`.